arangod/Pregel/Worker/Worker.cpp

```diff
@@ -594,8 +612,23 @@ auto Worker<V, E, M>::aqlResult(bool withId) const -> PregelResults {
   auto storer =
       GraphVPackBuilderStorer<V, E>(withId, _config, _algorithm->inputFormat(),
                                     std::move(_makeStatusCallback()));
-  // TODO GORDO-1599
-  storer.store(_quivers);
+  std::vector<futures::Future<Result>> storeFutures;
+  for (auto const& quiver : _quivers) {
+    storeFutures.emplace_back(storer.store({quiver}));
+  }
+  if (!storeFutures.empty()) {
+    // wait for all futures to finalize
+    auto futureResponses = futures::collectAll(storeFutures).get();
+    for (auto const& fResponse : futureResponses) {
+      auto& result = fResponse.get();
+      if (result.fail()) {
+        LOG_PREGEL("ee2af", WARN) << fmt::format(
+            "Thread failed to store graph results: {}", result.errorMessage());
+        // TODO: Anything else we can do in case we've failed here?!
+        throw;
```

Comment:
throw what? `throw;` which is written below make sense only in catch block.
It's rethrow